systemd/systemd.go

```diff
@@ -1091,9 +1143,28 @@ LazyUnmount=yes
 WantedBy=multi-user.target
 `
 
-func writeMountUnitFile(snapName, revision, what, where, fstype string, options []string) (mountUnitName string, err error) {
-	content := fmt.Sprintf(mountUnitTemplate, snapName, revision, what, where, fstype, strings.Join(options, ","))
-	mu := MountUnitPath(where)
+const (
+	snappyOriginModule = "X-SnapdOrigin"
+)
+
+func writeMountUnitFile(u *MountUnitOptions) (mountUnitName string, err error) {
+	if u == nil {
+		panic("writeMountUnitFile() expects valid mount options")
```

Comment:
Could this instead  be an error?  I am a bit worried about panics in snapd in general and given that we already return errors here at least snap would not crash and potentially run into an endless crash loop  (if this is part of a task snapd would keep trying to run the task) with an error instead of a panic?

Reply:
My reasoning was that passing `nil` for the option is a programming error, something that in C would be handled by an `assert()`: panic when running tests (or in debug mode), and invisible in production. But I see that golang has not such a thing, so I think the best option here is to follow your advice and return an error.

Reply:
Done :-)